eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -1115,11 +1105,13 @@ object Helpers {
       // are there remaining spendable outputs from the commitment tx?
       val commitOutputsSpendableByUs = (revokedCommitPublished.claimMainOutputTx.toSeq ++ revokedCommitPublished.mainPenaltyTx ++ revokedCommitPublished.htlcPenaltyTxs)
         .flatMap(_.txIn.map(_.outPoint)).toSet -- revokedCommitPublished.irrevocablySpent.keys
-      // which htlc delayed txes can we expect to be confirmed?
-      val unconfirmedHtlcDelayedTxes = revokedCommitPublished.claimHtlcDelayedPenaltyTxs
-        .filter(tx => (tx.txIn.map(_.outPoint.txid).toSet -- revokedCommitPublished.irrevocablySpent.values).isEmpty) // only the txes which parents are already confirmed may get confirmed (note that this also eliminates outputs that have been double-spent by a competing tx)
-        .filterNot(tx => revokedCommitPublished.irrevocablySpent.values.toSet.contains(tx.txid)) // has the tx already been confirmed?
-      isCommitTxConfirmed && commitOutputsSpendableByUs.isEmpty && unconfirmedHtlcDelayedTxes.isEmpty
+      // which htlc delayed txs can we expect to be confirmed?
+      val unconfirmedHtlcDelayedTxs = revokedCommitPublished.claimHtlcDelayedPenaltyTxs
+        // only the txs which parents are already confirmed may get confirmed (note that this also eliminates outputs that have been double-spent by a competing tx)
+        .filter(tx => (tx.txIn.map(_.outPoint.txid).toSet -- revokedCommitPublished.irrevocablySpent.values).isEmpty)
+        // if one of the tx inputs has been spent, the tx has already been confirmed or a competing tx has been confirmed
+        .filterNot(tx => tx.txIn.exists(txIn => revokedCommitPublished.irrevocablySpent.contains(txIn.outPoint)))
```

Comment:
This is giving me a headache. Aren't we changing the meaning of `irrevocablySpent`?

Reply:
I don't see why...one of our inputs has been irrevocably spent, so our tx is now obsolete and will never confirm. Can you detail?

Reply:
I thought that in all other cases the `irrevocablySpent` map only contained transactions that are in the `xxxPublished` object, but that's wrong: there are other instances where we have competing transactions, and we use that code:
```scala
  // does the tx spend an output of the local commitment tx?
  val spendsTheCommitTx = localCommitPublished.commitTx.txid == outPoint.txid
```

Reply:
Yes, we have to, we record in `irrevocablySpent` anything that confirms and consumes an outpoint we're interested in.
This is clearly a headache-heavy part of our codebase :)